lib/Adapters/RegressionAdapter/hlsStreamRegressionValidator.js

```diff
@@ -0,0 +1,231 @@
+/**
+ * Created by lilach.maliniak on 12/07/2016.
+ */
+var Q = require('q');
+var _ = require('underscore');
+var fs = require('fs');
+var util=require('util');
+var logger = require('../../../common/logger');
+var m3u8Handler = require('../../manifest/promise-m3u8');
+var mkdirp = require('mkdirp');
+var getDirName = require('path').dirname;
+var networkClient = require('./../../NetworkClientFactory').getNetworkClient();
+const crypto = require('crypto');
+const v_utility = require('./validatorUtility');
+
+function checksum (str, algorithm, encoding) {
+    return crypto
+        .createHash(algorithm || 'md5')
+        .update(str, 'utf8')
+        .digest(encoding || 'hex')
+}
+
+/*
+  ------------------------------------------------------------------------------------------------------
+  Class HlsStreamRegressionValidator
+  ------------------------------------------------------------------------------------------------------
+  This class is responsibilities:
+   (1) Generate baseline database (json file), with properties of stream (e.g. bytes count, checksum) for
+   all the chunklists and segments (ts files), fetched from the packager for single Live entry, during regression test
+   period.
+ */
+function HlsStreamRegressionValidator(entryId, flavors, last_chunklist_index, config) {
+
+    this.logger = logger.getLogger('hlsStreamRegressionValidator');
+    this.entryId = entryId;
+    this.flavors = flavors;
+    this.config = config;
+    this.override = config.override;
+    this.last_chunklist_index = last_chunklist_index;
+    // path to load/save regression test results
+    // the path is initialized in init()
+    this.fullpath = config.path;
+    // add root to path
+    var filename = this.entryId + '_regression_test_db.json';
+
+    if (this.fullpath.lastIndexOf('/') < this.fullpath.length - 1) {
+        this.fullpath += '/';
+        this.fullpath += filename;
+    }
+    
+    // create initial json for regression results database
+    this.regression_results_db;
+    this.this_run_results = {};
+
+}
+
+HlsStreamRegressionValidator.prototype.init = function() {
+
+
+    var that = this;
+    return this.loadResultsDatabase()
+        .then( function() {
+            that.logger.debug('[%s] regression tests validator successfully initialized.', that.entryId);
+        })
+        .catch (function(err) {
+            that.logger.error('[%s] failed to initialize regression tests validator. Error: %s, stack: %s', that.entryId, err.message, err.stack);
+        });
+}
+
+HlsStreamRegressionValidator.prototype.createDestinationDir = function() {
```

Comment:
should +HlsStreamRegressionValidator.prototype.createDestinationDir  be in utils?